File: 2000 98 & ME/QuickStart/winforms/samples/VB/Data/SimpleBinding/SimpleBinding.vb
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.WinForms

Imports Microsoft.Samples.WinForms.VB.SimpleBinding.Data

Namespace Microsoft.Samples.WinForms.VB.SimpleBinding 

    Public Class SimpleBinding
        Inherits System.WinForms.Form

        'Required by the Windows Forms Designer   
        Private components As System.ComponentModel.Container
        Private labelTitle As System.WinForms.Label
        Private buttonMovePrev As System.WinForms.Button
        Private textBoxLastName As System.WinForms.TextBox
        Private textBoxPosition As System.WinForms.TextBox
        Private labelDOB As System.WinForms.Label
        Private labelFirstName As System.WinForms.Label
        Private textBoxTitle As System.WinForms.TextBox
        Private dateTimeFormat1 As System.WinForms.ComponentModel.DateTimeFormat
        Private buttonMoveNext As System.WinForms.Button
        Private textBoxID As System.WinForms.TextBox
        Private buttonMoveFirst As System.WinForms.Button
        Private labelID As System.WinForms.Label
        Private labelAddress As System.WinForms.Label
        Private buttonMoveLast As System.WinForms.Button
        Private textBoxFirstName As System.WinForms.TextBox
        Private textBoxDOB As System.WinForms.TextBox
        Private panelVCRControl As System.WinForms.Panel
        Private labelLastName As System.WinForms.Label
        Private textBoxAddress As System.WinForms.TextBox
        
	    Private custList() As Customer
    
        Public Sub New()
           MyBase.New
    
           'This call is required by the Windows Forms Designer.
           InitializeComponent
    
           ' TODO: Add any constructor code after InitializeComponent call

            ' Get the list of customers as an array
            custList = CustomerList.GetCustomersArray()

            ' Set up the bindings so that each field on the form is
            ' bound to a property of Customer
            textBoxID.Bindings.Add("Text", custList, "ID")
            textBoxTitle.Bindings.Add("Text", custList, "Title")
            textBoxLastName.Bindings.Add("Text", custList, "LastName")
            textBoxFirstName.Bindings.Add("Text", custList, "FirstName")
            textBoxDOB.Bindings.Add("Value", custList, "DateOfBirth")
            textBoxAddress.Bindings.Add("Text", custList, "Address")

            ' We want to handle position changing events for the DATA VCR Panel
            ' Position is managed by the Form's BindingManager so hook the position changed
            ' event on the BindingManager
            AddHandler Me.BindingManager(custList).PositionChanged, new System.EventHandler(AddressOf customers_PositionChanged)
            
            ' Set up the initial text for the DATA VCR Panel
            textBoxPosition.Text = "Record " + (Me.BindingManager(custList).Position + 1).ToString() + " of " + custList.Length.ToString()
            
            ' Set the minimum form size to the client size + the height of the title bar
            Me.MinTrackSize = new Size(368, (413 + SystemInformation.CaptionHeight))

        End Sub

        ' When the MoveFirst button is clicked set the position for the CustomersList
        ' to the first record
        private Sub buttonMoveFirst_Click(ByVal sender As Object, ByVal e As System.EventArgs) 
            Me.BindingManager(custList).Position = 0 
        End Sub


        ' When the MoveLast button is clicked set the position for the CustomersList 
        ' to the last record
        private Sub buttonMoveLast_Click(ByVal sender As Object, ByVal e As System.EventArgs) 
            Me.BindingManager(custList).Position = custList.Length - 1
        End Sub


        ' When the MoveNext button is clicked increment the position for the CustomersList
        private Sub buttonMoveNext_Click(ByVal sender As Object, ByVal e As System.EventArgs) 
            If Me.BindingManager(custList).Position < custList.Length - 1 Then 
                Me.BindingManager(custList).Position = Me.BindingManager(custList).Position + 1
            End if
        End Sub


        ' When the MovePrev button is clicked decrement the position for the CustomersList
        private Sub buttonMovePrev_Click(ByVal sender As Object, ByVal e As System.EventArgs) 
            if Me.BindingManager(custList).Position > 0 Then 
                Me.BindingManager(custList).Position -=	1
            End If
        End Sub


        ' Position has changed - update the DATA VCR panel
        private Sub customers_PositionChanged(ByVal sender As Object, ByVal e As System.EventArgs) 
            textBoxPosition.Text = "Record " + (Me.BindingManager(custList).Position + 1).ToString() + " of " + custList.Length.ToString()
        End Sub


        'Clean up any resources being used
        Overrides Public Sub Dispose()
            MyBase.Dispose
            components.Dispose
        End Sub 

        'The main entry point for the application
        Shared Sub Main()
            System.WinForms.Application.Run(New SimpleBinding())
        End Sub

        'NOTE: The following procedure is required by the Windows Forms Designer
        'Do not modify it.
        Private Sub InitializeComponent() 
            Me.components = New System.ComponentModel.Container
            Me.textBoxTitle = New System.WinForms.TextBox
            Me.labelFirstName = New System.WinForms.Label
            Me.dateTimeFormat1 = New System.WinForms.ComponentModel.DateTimeFormat
            Me.buttonMoveNext = New System.WinForms.Button
            Me.buttonMovePrev = New System.WinForms.Button
            Me.labelTitle = New System.WinForms.Label
            Me.textBoxLastName = New System.WinForms.TextBox
            Me.labelDOB = New System.WinForms.Label
            Me.textBoxPosition = New System.WinForms.TextBox
            Me.textBoxID = New System.WinForms.TextBox
            Me.textBoxFirstName = New System.WinForms.TextBox
            Me.textBoxDOB = New System.WinForms.TextBox
            Me.textBoxAddress = New System.WinForms.TextBox
            Me.panelVCRControl = New System.WinForms.Panel
            Me.buttonMoveFirst = New System.WinForms.Button
            Me.labelLastName = New System.WinForms.Label
            Me.labelID = New System.WinForms.Label
            Me.buttonMoveLast = New System.WinForms.Button
            Me.labelAddress = New System.WinForms.Label

            textBoxTitle.Location = New System.Drawing.Point(88, 70)
            textBoxTitle.Text = ""
            textBoxTitle.TabIndex = 1
            textBoxTitle.Size = New System.Drawing.Size(70, 20)

            labelFirstName.Location = New System.Drawing.Point(8, 112)
            labelFirstName.Text = "&First Name:"
            labelFirstName.Size = New System.Drawing.Size(64, 16)
            labelFirstName.TabIndex = 7

            dateTimeFormat1.DisplayFormat = "D"
            '@design dateTimeFormat1.SetLocation(New System.Drawing.Point(7, 7))

            Me.AutoScaleBaseSize = New System.Drawing.Size(5, 13)
            Me.Text = "Customer Details"
            '@design Me.TrayLargeIcon = True
            '@design Me.TrayHeight = 93
            Me.ClientSize = New System.Drawing.Size(368, 413)

            buttonMoveNext.Location = New System.Drawing.Point(184, 8)
            buttonMoveNext.FlatStyle = System.WinForms.FlatStyle.Flat
            buttonMoveNext.Size = New System.Drawing.Size(32, 32)
            buttonMoveNext.TabIndex = 11
            buttonMoveNext.Anchor = System.WinForms.AnchorStyles.Top
            buttonMoveNext.Font = New System.Drawing.Font("Tahoma", 11!, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.World)
            buttonMoveNext.Text = ">"
            AddHandler buttonMoveNext.Click, New System.EventHandler(AddressOf Me.buttonMoveNext_Click)

            buttonMovePrev.Location = New System.Drawing.Point(48, 8)
            buttonMovePrev.FlatStyle = System.WinForms.FlatStyle.Flat
            buttonMovePrev.Size = New System.Drawing.Size(32, 32)
            buttonMovePrev.TabIndex = 12
            buttonMovePrev.Font = New System.Drawing.Font("Tahoma", 11!, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.World)
            buttonMovePrev.Text = "<"
            AddHandler buttonMovePrev.Click, New System.EventHandler(AddressOf Me.buttonMovePrev_Click)

            labelTitle.Location = New System.Drawing.Point(8, 72)
            labelTitle.Text = "&Title:"
            labelTitle.Size = New System.Drawing.Size(64, 16)
            labelTitle.TabIndex = 6

            textBoxLastName.Location = New System.Drawing.Point(88, 152)
            textBoxLastName.Text = ""
            textBoxLastName.Anchor = System.WinForms.AnchorStyles.TopLeftRight
            textBoxLastName.TabIndex = 3
            textBoxLastName.Size = New System.Drawing.Size(243, 20)

            labelDOB.Location = New System.Drawing.Point(8, 194)
            labelDOB.Text = "&Date of Birth:"
            labelDOB.Size = New System.Drawing.Size(72, 16)
            labelDOB.TabIndex = 16

            textBoxPosition.Location = New System.Drawing.Point(88, 14)
            textBoxPosition.Text = ""
            textBoxPosition.Anchor = System.WinForms.AnchorStyles.Left
            textBoxPosition.BorderStyle = System.WinForms.BorderStyle.FixedSingle
            textBoxPosition.ReadOnly = True
            textBoxPosition.TabIndex = 14
            textBoxPosition.Size = New System.Drawing.Size(88, 20)

            textBoxID.Location = New System.Drawing.Point(88, 30)
            textBoxID.Text = ""
            textBoxID.ReadOnly = True
            textBoxID.TabIndex = 0
            textBoxID.Size = New System.Drawing.Size(203, 20)

            textBoxFirstName.Location = New System.Drawing.Point(88, 112)
            textBoxFirstName.Text = ""
            textBoxFirstName.Anchor = System.WinForms.AnchorStyles.TopLeftRight
            textBoxFirstName.TabIndex = 2
            textBoxFirstName.Size = New System.Drawing.Size(243, 20)

            textBoxDOB.Location = New System.Drawing.Point(88, 192)
            textBoxDOB.Anchor = System.WinForms.AnchorStyles.TopLeftRight
            textBoxDOB.Format = dateTimeFormat1
            textBoxDOB.TabIndex = 17
            textBoxDOB.Size = New System.Drawing.Size(243, 20)

            textBoxAddress.Location = New System.Drawing.Point(88, 232)
            textBoxAddress.Text = ""
            textBoxAddress.Anchor = System.WinForms.AnchorStyles.All
            textBoxAddress.Multiline = True
            textBoxAddress.AcceptsReturn = True
            textBoxAddress.TabIndex = 4
            textBoxAddress.Size = New System.Drawing.Size(264, 88)

            panelVCRControl.BorderStyle = System.WinForms.BorderStyle.FixedSingle
            panelVCRControl.Location = New System.Drawing.Point(88, 344)
            panelVCRControl.Size = New System.Drawing.Size(264, 48)
            panelVCRControl.TabIndex = 15
            panelVCRControl.Anchor = System.WinForms.AnchorStyles.Bottom
            panelVCRControl.Text = "panel1"

            buttonMoveFirst.Location = New System.Drawing.Point(8, 8)
            buttonMoveFirst.FlatStyle = System.WinForms.FlatStyle.Flat
            buttonMoveFirst.Size = New System.Drawing.Size(32, 32)
            buttonMoveFirst.TabIndex = 13
            buttonMoveFirst.Font = New System.Drawing.Font("Tahoma", 11!, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.World)
            buttonMoveFirst.Text = "|<"
            AddHandler buttonMoveFirst.Click, New System.EventHandler(AddressOf Me.buttonMoveFirst_Click)

            labelLastName.Location = New System.Drawing.Point(8, 154)
            labelLastName.Text = "&Last Name:"
            labelLastName.Size = New System.Drawing.Size(64, 16)
            labelLastName.TabIndex = 8

            labelID.Location = New System.Drawing.Point(8, 32)
            labelID.Text = "ID:"
            labelID.Size = New System.Drawing.Size(64, 16)
            labelID.TabIndex = 5

            buttonMoveLast.Location = New System.Drawing.Point(224, 8)
            buttonMoveLast.FlatStyle = System.WinForms.FlatStyle.Flat
            buttonMoveLast.Size = New System.Drawing.Size(32, 32)
            buttonMoveLast.TabIndex = 10
            buttonMoveLast.Anchor = System.WinForms.AnchorStyles.Top
            buttonMoveLast.Font = New System.Drawing.Font("Tahoma", 11!, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.World)
            buttonMoveLast.Text = ">|"
            AddHandler buttonMoveLast.Click, New System.EventHandler(AddressOf Me.buttonMoveLast_Click)

            labelAddress.Location = New System.Drawing.Point(8, 232)
            labelAddress.Text = "&Address:"
            labelAddress.Size = New System.Drawing.Size(64, 16)
            labelAddress.TabIndex = 9

            panelVCRControl.Controls.Add(textBoxPosition)
            panelVCRControl.Controls.Add(buttonMoveFirst)
            panelVCRControl.Controls.Add(buttonMovePrev)
            panelVCRControl.Controls.Add(buttonMoveNext)
            panelVCRControl.Controls.Add(buttonMoveLast)
            Me.Controls.Add(textBoxAddress)
            Me.Controls.Add(labelLastName)
            Me.Controls.Add(panelVCRControl)
            Me.Controls.Add(textBoxDOB)
            Me.Controls.Add(textBoxFirstName)
            Me.Controls.Add(labelAddress)
            Me.Controls.Add(labelID)
            Me.Controls.Add(textBoxID)
            Me.Controls.Add(textBoxTitle)
            Me.Controls.Add(labelFirstName)
            Me.Controls.Add(labelDOB)
            Me.Controls.Add(textBoxLastName)
            Me.Controls.Add(labelTitle)

        End Sub

    End Class

End Namespace

File: 2000 98 & ME/QuickStart/winforms/samples/Cs/Data/SimpleBinding/SimpleBinding.cs
//------------------------------------------------------------------------------
/// <copyright from='1997' to='2001' company='Microsoft Corporation'>           
///    Copyright (c) Microsoft Corporation. All Rights Reserved.                
///       
///    This source code is intended only as a supplement to Microsoft
///    Development Tools and/or on-line documentation.  See these other
///    materials for detailed information regarding Microsoft code samples.
///
/// </copyright>                                                                
//------------------------------------------------------------------------------
namespace Microsoft.Samples.WinForms.Cs.SimpleBinding {
    using System;
    using System.ComponentModel;
    using System.Drawing;
    using System.WinForms;

    using System.Data;
    using System.Data.ADO;
    using Microsoft.Samples.WinForms.Cs.SimpleBinding.Data;

    public class SimpleBinding : System.WinForms.Form {
        private System.ComponentModel.Container components;
    
    	private System.WinForms.ComponentModel.DateTimeFormat dateTimeFormat1;
	    private System.WinForms.TextBox textBoxDOB;
	    private System.WinForms.Label labelDOB;
	    private System.WinForms.TextBox textBoxPosition;
	    private System.WinForms.Button buttonMoveFirst;
	    private System.WinForms.Button buttonMovePrev;
	    private System.WinForms.Button buttonMoveNext;
	    private System.WinForms.Button buttonMoveLast;
        private System.WinForms.TextBox textBoxTitle;
        private System.WinForms.Label labelTitle;
        private System.WinForms.TextBox textBoxAddress;
        private System.WinForms.TextBox textBoxLastName;
        private System.WinForms.TextBox textBoxFirstName;
        private System.WinForms.TextBox textBoxID;
        private System.WinForms.Label labelAddress;
        private System.WinForms.Label labelLastName;
        private System.WinForms.Label labelFirstName;
        private System.WinForms.Label labelID;
        private System.WinForms.Panel panelVCRControl;
        private CustomerList custList;

        public SimpleBinding() {
            
            // Required by the Windows Forms Designer
            InitializeComponent();

            // TODO: Add any constructor code after InitializeComponent call

            //Get the list of customers
            custList = CustomerList.GetCustomers();

            //Set up the bindings so that each field on the form is
            //bound to a property of Customer
            textBoxID.Bindings.Add("Text", custList, "ID");
            textBoxTitle.Bindings.Add("Text", custList, "Title");
            textBoxLastName.Bindings.Add("Text", custList, "LastName");
            textBoxFirstName.Bindings.Add("Text", custList, "FirstName");
            textBoxDOB.Bindings.Add("Value", custList, "DateOfBirth");
            textBoxAddress.Bindings.Add("Text", custList, "Address");

            //We want to handle position changing events for the DATA VCR Panel
            //Position is managed by the Form's BindingManager so hook the position changed
            //event on the BindingManager
            this.BindingManager[custList].PositionChanged += new System.EventHandler(customers_PositionChanged);
            
            //Set up the initial text for the DATA VCR Panel
            textBoxPosition.Text = "Record " + (this.BindingManager[custList].Position + 1) + " of " + custList.Count;
            
            //Set the minimum form size to the client size + the height of the title bar
            this.MinTrackSize = new Size(368, (413 + SystemInformation.CaptionHeight));

        }


        //When the MoveFirst button is clicked set the position for the CustomersList
        //to the first record
        private void buttonMoveFirst_Click(object sender, System.EventArgs e) {
            this.BindingManager[custList].Position = 0 ;
        }


        //When the MoveLast button is clicked set the position for the CustomersList 
        //to the last record
        private void buttonMoveLast_Click(object sender, System.EventArgs e) {
            this.BindingManager[custList].Position = custList.Count - 1;
        }


        //When the MoveNext button is clicked increment the position for the CustomersList
        private void buttonMoveNext_Click(object sender, System.EventArgs e) {
            if (this.BindingManager[custList].Position < custList.Count - 1) {
                this.BindingManager[custList].Position++;
            }
        }


        //When the MovePrev button is clicked decrement the position for the CustomersList
        private void buttonMovePrev_Click(object sender, System.EventArgs e) {
            if (this.BindingManager[custList].Position > 0) {
                this.BindingManager[custList].Position--;
            }
        }


        //Position has changed - update the DATA VCR panel
        private void customers_PositionChanged(object sender, System.EventArgs e) {
            textBoxPosition.Text = "Record " + (this.BindingManager[custList].Position + 1) + " of " + custList.Count;
        }


        public override void Dispose() {
            base.Dispose();
            components.Dispose();
        }


        private void InitializeComponent()
	{
		this.components = new System.ComponentModel.Container();
		this.textBoxTitle = new System.WinForms.TextBox();
		this.labelFirstName = new System.WinForms.Label();
		this.dateTimeFormat1 = new System.WinForms.ComponentModel.DateTimeFormat();
		this.textBoxID = new System.WinForms.TextBox();
		this.buttonMoveNext = new System.WinForms.Button();
		this.buttonMovePrev = new System.WinForms.Button();
		this.labelTitle = new System.WinForms.Label();
		this.textBoxLastName = new System.WinForms.TextBox();
		this.labelDOB = new System.WinForms.Label();
		this.textBoxPosition = new System.WinForms.TextBox();
		this.textBoxDOB = new System.WinForms.TextBox();
		this.panelVCRControl = new System.WinForms.Panel();
		this.textBoxAddress = new System.WinForms.TextBox();
		this.labelLastName = new System.WinForms.Label();
		this.labelID = new System.WinForms.Label();
		this.buttonMoveFirst = new System.WinForms.Button();
		this.labelAddress = new System.WinForms.Label();
		this.textBoxFirstName = new System.WinForms.TextBox();
		this.buttonMoveLast = new System.WinForms.Button();
		
		//@design this.TrayHeight = 181;
		//@design this.TrayLargeIcon = true;
		//@design this.TrayAutoArrange = true;
		this.Text = "Customer Details";
		this.AutoScaleBaseSize = new System.Drawing.Size(5, 13);
		this.ClientSize = new System.Drawing.Size(368, 413);
		
		textBoxTitle.Location = new System.Drawing.Point(88, 70);
		textBoxTitle.TabIndex = 1;
		textBoxTitle.Size = new System.Drawing.Size(70, 20);
		
		labelFirstName.Location = new System.Drawing.Point(8, 112);
		labelFirstName.Text = "&First Name:";
		labelFirstName.Size = new System.Drawing.Size(64, 16);
		labelFirstName.TabIndex = 7;
		
		//@design dateTimeFormat1.SetLocation(new System.Drawing.Point(7, 7));
		dateTimeFormat1.DisplayFormat = "D";
		
		textBoxID.Location = new System.Drawing.Point(88, 30);
		textBoxID.ReadOnly = true;
		textBoxID.Enabled = false;
		textBoxID.TabIndex = 0;
		textBoxID.Size = new System.Drawing.Size(203, 20);
		
		buttonMoveNext.Location = new System.Drawing.Point(184, 8);
		buttonMoveNext.FlatStyle = System.WinForms.FlatStyle.Flat;
		buttonMoveNext.Size = new System.Drawing.Size(32, 32);
		buttonMoveNext.TabIndex = 11;
		buttonMoveNext.Anchor = System.WinForms.AnchorStyles.Top;
		buttonMoveNext.Font = new System.Drawing.Font("Times New Roman", 11f, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.World);
		buttonMoveNext.Text = ">";
		buttonMoveNext.Click += new System.EventHandler(buttonMoveNext_Click);
		
		buttonMovePrev.Location = new System.Drawing.Point(48, 8);
		buttonMovePrev.FlatStyle = System.WinForms.FlatStyle.Flat;
		buttonMovePrev.Size = new System.Drawing.Size(32, 32);
		buttonMovePrev.TabIndex = 12;
		buttonMovePrev.Font = new System.Drawing.Font("Times New Roman", 11f, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.World);
		buttonMovePrev.Text = "<";
		buttonMovePrev.Click += new System.EventHandler(buttonMovePrev_Click);
		
		labelTitle.Location = new System.Drawing.Point(8, 72);
		labelTitle.Text = "&Title:";
		labelTitle.Size = new System.Drawing.Size(64, 16);
		labelTitle.TabIndex = 6;
		
		textBoxLastName.Location = new System.Drawing.Point(88, 152);
		textBoxLastName.TabIndex = 3;
		textBoxLastName.Anchor = System.WinForms.AnchorStyles.TopLeftRight;
		textBoxLastName.Size = new System.Drawing.Size(243, 20);
		
		labelDOB.Location = new System.Drawing.Point(8, 194);
		labelDOB.Text = "&Date of Birth:";
		labelDOB.Size = new System.Drawing.Size(72, 16);
		labelDOB.TabIndex = 16;
		
		textBoxPosition.Location = new System.Drawing.Point(88, 14);
		textBoxPosition.ReadOnly = true;
		textBoxPosition.BorderStyle = System.WinForms.BorderStyle.FixedSingle;
		textBoxPosition.Enabled = false;
		textBoxPosition.TabIndex = 14;
		textBoxPosition.Anchor = System.WinForms.AnchorStyles.Left;
		textBoxPosition.Size = new System.Drawing.Size(88, 20);
		
		textBoxDOB.Location = new System.Drawing.Point(88, 192);
		textBoxDOB.Format = dateTimeFormat1;
		textBoxDOB.TabIndex = 17;
		textBoxDOB.Anchor = System.WinForms.AnchorStyles.TopLeftRight;
		textBoxDOB.Size = new System.Drawing.Size(243, 20);
		
		panelVCRControl.BorderStyle = System.WinForms.BorderStyle.FixedSingle;
		panelVCRControl.Location = new System.Drawing.Point(88, 344);
		panelVCRControl.Size = new System.Drawing.Size(264, 48);
		panelVCRControl.TabIndex = 15;
		panelVCRControl.Anchor = System.WinForms.AnchorStyles.Bottom;
		panelVCRControl.Text = "panel1";
		
		textBoxAddress.Location = new System.Drawing.Point(88, 232);
		textBoxAddress.Multiline = true;
		textBoxAddress.AcceptsReturn = true;
		textBoxAddress.TabIndex = 4;
		textBoxAddress.Anchor = System.WinForms.AnchorStyles.All;
		textBoxAddress.Size = new System.Drawing.Size(264, 88);
		
		labelLastName.Location = new System.Drawing.Point(8, 154);
		labelLastName.Text = "&Last Name:";
		labelLastName.Size = new System.Drawing.Size(64, 16);
		labelLastName.TabIndex = 8;
		
		labelID.Location = new System.Drawing.Point(8, 32);
		labelID.Text = "ID:";
		labelID.Size = new System.Drawing.Size(64, 16);
		labelID.TabIndex = 5;
		
		buttonMoveFirst.Location = new System.Drawing.Point(8, 8);
		buttonMoveFirst.FlatStyle = System.WinForms.FlatStyle.Flat;
		buttonMoveFirst.Size = new System.Drawing.Size(32, 32);
		buttonMoveFirst.TabIndex = 13;
		buttonMoveFirst.Font = new System.Drawing.Font("Times New Roman", 11f, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.World);
		buttonMoveFirst.Text = "|<";
		buttonMoveFirst.Click += new System.EventHandler(buttonMoveFirst_Click);
		
		labelAddress.Location = new System.Drawing.Point(8, 232);
		labelAddress.Text = "&Address:";
		labelAddress.Size = new System.Drawing.Size(64, 16);
		labelAddress.TabIndex = 9;
		
		textBoxFirstName.Location = new System.Drawing.Point(88, 112);
		textBoxFirstName.TabIndex = 2;
		textBoxFirstName.Anchor = System.WinForms.AnchorStyles.TopLeftRight;
		textBoxFirstName.Size = new System.Drawing.Size(243, 20);
		
		buttonMoveLast.Location = new System.Drawing.Point(224, 8);
		buttonMoveLast.FlatStyle = System.WinForms.FlatStyle.Flat;
		buttonMoveLast.Size = new System.Drawing.Size(32, 32);
		buttonMoveLast.TabIndex = 10;
		buttonMoveLast.Anchor = System.WinForms.AnchorStyles.Top;
		buttonMoveLast.Font = new System.Drawing.Font("Times New Roman", 11f, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.World);
		buttonMoveLast.Text = ">|";
		buttonMoveLast.Click += new System.EventHandler(buttonMoveLast_Click);
		panelVCRControl.Controls.Add(textBoxPosition);
		panelVCRControl.Controls.Add(buttonMoveFirst);
		panelVCRControl.Controls.Add(buttonMovePrev);
		panelVCRControl.Controls.Add(buttonMoveNext);
		panelVCRControl.Controls.Add(buttonMoveLast);
		this.Controls.Add(textBoxDOB);
		this.Controls.Add(labelDOB);
		this.Controls.Add(panelVCRControl);
		this.Controls.Add(textBoxTitle);
		this.Controls.Add(labelTitle);
		this.Controls.Add(textBoxAddress);
		this.Controls.Add(textBoxLastName);
		this.Controls.Add(textBoxFirstName);
		this.Controls.Add(textBoxID);
		this.Controls.Add(labelAddress);
		this.Controls.Add(labelLastName);
		this.Controls.Add(labelFirstName);
		this.Controls.Add(labelID);
	}


        public static void Main(string[] args) {
            Application.Run(new SimpleBinding());
        }

    }
}
















